torchchat/generate.py

```diff
@@ -531,15 +531,22 @@ def decode_n_tokens(
         callback=lambda _: _,
         eos_token_id: int = 2,
         eot_id: Optional[int] = None,
+        attention_backend: str = "math",
         **sampling_kwargs,
     ):
         new_tokens, new_probs = [], []
         encountered_eos = False
+        sdp_backend_dict = {
+            'math': torch.nn.attention.SDPBackend.MATH,
+            'flash_attention': torch.nn.attention.SDPBackend.FLASH_ATTENTION,
+            'efficient_attention': torch.nn.attention.SDPBackend.EFFICIENT_ATTENTION,
+            'cudnn_attention': torch.nn.attention.SDPBackend.CUDNN_ATTENTION,
+        }
```

Comment:
Can we push this into the builder.py as well?

Reply:
Done.